prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -218,6 +219,84 @@ def canonicalize_dataset_using_columns(
             )
         return datasets.DatasetDict(dataset_dict)
 
+    def canonicalize_dataset_auto(
+        self, dataset_name: str, prompt_spec: PromptSpec, num_transform: int = 3000
```

Comment:
I don't think that the variable name `num_transform` is not sufficiently descriptive. Can we rename it to `num_points_to_transform`? 

The corresponding docstring is also not detailed enough and needs more elaboration (e.g. what happens if the dataset size is larger than  `num_points_to_transform`?). The docstring below, "number of data points you wish to transform" is a good starting point (and more informative than the one shown here)